sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/io/Providers.java

```diff
@@ -44,19 +48,19 @@ public static <T extends Identifyable> Map<String, T> loadProviders(Class<T> kla
     for (T provider : ServiceLoader.load(klass)) {
       // Avro provider is treated as a special case since two Avro providers may want to be loaded -
       // from "core" (deprecated) and from "extensions/avro" (actual) - but only one must succeed.
-      // TODO: this check should be removed once once AvroPayloadSerializerProvider from "core" is
+      // TODO: we won't need this check once all Avro providers from "core" will be
       // removed
       if (provider.identifier().equals("avro")) {
         // Avro provider from "extensions/avro" must have a priority.
-        if (provider
-            .getClass()
-            .getName()
-            .equals(
-                "org.apache.beam.sdk.extensions.avro.schemas.io.payloads.AvroPayloadSerializerProvider")) {
-          // Use AvroPayloadSerializerProvider from extensions/avro by any case.
+        if (provider.getClass().getName().startsWith("org.apache.beam.sdk.extensions.avro")) {
+          // Load Avro provider from "extensions/avro" by any case.
           providers.put(provider.identifier(), provider);
         } else {
           // Load Avro provider from "core" if it was not loaded from Avro extension before.
+          LOG.warn(
```

Comment:
Unfortunately this is generating noise and might warn users even if the Avro extension is on the classpath depending on the order :/
I think it's better to keep this as a separate task @aromanenko-dev. Adding warning logs isn't urgent at all.